src/hoarder/archives/rar_path.py

```
@@ -179,3 +182,75 @@ def find_rar_files(
+        main_volume = rar_volumes[0]
+        logger.debug("Main volume is %s", main_volume)
+    else:
+        logger.debug("Path %s is not a file or directory", full_path)
+        raise FileNotFoundError(f"{full_path} could not be found")
+
+    try:
+        main_volume_path = main_volume.relative_to(storage_path)
+    except ValueError:
+        raise ValueError(
+            f"Main volume {main_volume} is not under storage_path {storage_path}"
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)